linera-chain/src/certificate.rs

```diff
@@ -255,15 +267,25 @@ impl<T: Clone> Clone for Hashed<T> {
 }
 
 impl<T> GenericCertificate<T> {
-    pub fn new(
-        value: T,
-        old_hash: CryptoHash,
+    pub fn unchecked_new(
```

Comment:
I was a bit confused by that name at first: I wouldn't _expect_ this function to check the hash, because at this point the `Hashed` has already been created (in a checked or unchecked way). In the context of a _certificate_ I would interpret the word "check" with verifying the _signatures_.

Reply:
I'll remove this constructor and leave the `new(value: T)` only. And define a `fn hash(&self)` on an instance where `T = Hashed<T>`.

Reply:
Although this might get a bit messy since almost everywhere we assume that `value` is already wrapped with `Hashed<_>` type. I'll give it a go and see if it works.

Reply:
See https://github.com/linera-io/linera-protocol/pull/2877#discussion_r1840561848.